apps/meteor/server/services/omnichannel-verification/service.ts

```diff
@@ -63,30 +65,37 @@ export class OmnichannelVerification extends ServiceClassInternal implements IOm
 		});
 	}
 
-	private async wrongInput(room: IOmnichannelGenericRoom, _isWrongOTP: boolean): Promise<boolean> {
+	private async wrongInput(room: IOmnichannelRoom, _isWrongOTP: boolean): Promise<boolean> {
 		const limitWrongAttempts: number = settings?.get('Livechat_LimitWrongAttempts') ?? 3;
 		if (!room?.wrongMessageCount) {
 			await LivechatRooms.updateWrongMessageCount(room._id, 1);
-		} else if (room.wrongMessageCount + 1 >= limitWrongAttempts) {
+			return true;
+		}
+		if (room.wrongMessageCount + 1 >= limitWrongAttempts) {
+			const { emailCode } = room.services || {};
 			await Promise.all([
 				LivechatRooms.updateWrongMessageCount(room._id, 0),
 				LivechatRooms.updateVerificationStatusById(room._id, RoomVerificationState.verifiedFalse),
+				emailCode && _isWrongOTP
+					? Promise.all(emailCode.map(({ code }) => LivechatRooms.removeEmailCodeByRoomIdAndCode(room._id, code)))
+					: Promise.resolve(),
```

Comment:
```suggestion
                ...[
					emailCode && _isWrongOTP
						? emailCode.map(({ code }) => LivechatRooms.removeEmailCodeByRoomIdAndCode(room._id, code))
						: Promise.resolve(),
				],
```
Didn't like this pattern tbh of Promise.all inside another promise.all. How about something like this?